File: Examples/Python/tests/testCore.py
from FMT import Core


def testCoreEnumsAreExposed():
    assert Core.FMTsection.Control is not None
    assert Core.FMToutputlevel.totalonly is not None
    assert Core.FMTconstrainttype.FMTstandard is not None


def testCoreDefaultObjectsCanBeConstructed():
    constructors = {
        "FMTmask": Core.FMTmask,
        "FMTdevelopment": Core.FMTdevelopment,
        "FMTagebounds": Core.FMTagebounds,
        "FMTperbounds": Core.FMTperbounds,
        "FMTyldbounds": Core.FMTyldbounds,
        "FMTlockbounds": Core.FMTlockbounds,
        "FMTspec": Core.FMTspec,
        "FMTactualdevelopment": Core.FMTactualdevelopment,
        "FMTaction": Core.FMTaction,
        "FMTlifespans": Core.FMTlifespans,
        "FMTfork": Core.FMTfork,
        "FMTyieldhandler": Core.FMTyieldhandler,
        "FMTyields": Core.FMTyields,
        "FMTtransition": Core.FMTtransition,
        "FMToperator": Core.FMToperator,
        "FMToutput": Core.FMToutput,
        "FMTtheme": Core.FMTtheme,
        "FMTschedule": Core.FMTschedule,
        "FMTconstraint": Core.FMTconstraint,
    }
    for name, constructor in constructors.items():
        assert constructor() is not None


def testSchedulePeriodRoundTrip():
    schedule = Core.FMTschedule()
    schedule.setperiod(3)
    assert schedule.getperiod() == 3

File: Examples/R/tests/testCore.R
test_that("Core exports construct and preserve values", {
  classNames <- c(
    "FMTobject", "FMTmask", "FMTmaskfilter", "FMTdevelopment",
    "FMTagebounds", "FMTperbounds", "FMTyldbounds", "FMTlockbounds",
    "FMTspec", "FMTactualdevelopment", "FMTfuturdevelopment",
    "FMTaction", "FMTlifespans", "FMTtransitionmask", "FMTfork",
    "FMTyieldhandler", "FMTyields", "FMTtransition", "FMToutputsource",
    "FMToperator", "FMToutputnode", "FMToutput", "FMTtheme",
    "FMTconstants", "FMTschedule", "FMTGCBMtransition", "FMTconstraint"
  )
  for (className in classNames) {
    expect_true(exists(className, inherits = TRUE))
  }

  schedule <- new(FMTschedule)
  schedule$setperiod(3L)
  expect_equal(schedule$getperiod(), 3L)
})
